demo/test.c

```diff
@@ -2219,54 +2224,61 @@ int unit_tests(int argc, char **argv)
       const char *name;
       int (*fn)(void);
    } test[] = {
-#define T(n) { #n, test_##n }
-      T(feature_detection),
-      T(trivial_stuff),
-      T(mp_get_set_i32),
-      T(mp_get_set_i64),
-      T(mp_and),
-      T(mp_cnt_lsb),
-      T(mp_complement),
-      T(mp_decr),
-      T(mp_div_3),
-      T(mp_dr_reduce),
-      T(mp_fread_fwrite),
-      T(mp_get_u32),
-      T(mp_get_u64),
-      T(mp_get_ul),
-      T(mp_ilogb),
-      T(mp_incr),
-      T(mp_invmod),
-      T(mp_is_square),
-      T(mp_kronecker),
-      T(mp_montgomery_reduce),
-      T(mp_root_u32),
-      T(mp_or),
-      T(mp_prime_is_prime),
-      T(mp_prime_next_prime),
-      T(mp_prime_rand),
-      T(mp_rand),
-      T(mp_read_radix),
-      T(mp_reduce_2k),
-      T(mp_reduce_2k_l),
+#define T_OFF           {"", NULL}
+#define T(n, o)         { #n, MP_HAS(o) ? test_##n : NULL }
+#define T2(n, o1, o2)   { #n, MP_HAS(o1) && MP_HAS(o2) ? test_##n : NULL }
+      T(feature_detection, TEST_FEATURE_DETECTION),
+      T(trivial_stuff, TEST_TRIVIAL_STUFF),
```

Comment:
Instead define T0, T1 and T2? Besides that it looks good.

Reply:
done!
now I've only to find out why that single build fails...